Doesn't compile with libb64 1.2.1 due to missing definition

Hello,
I have been trying to compile DCCL w/ b64 support and have come across an issue:
```
In file included from /tmp/dccl-20190311-2927-10tibqz/dccl-3.0.10/build/include/dccl/binary.h:34,
                 from /tmp/dccl-20190311-2927-10tibqz/dccl-3.0.10/build/include/dccl/codec.h:36,
                 from /tmp/dccl-20190311-2927-10tibqz/dccl-3.0.10/src/bitset.cpp:23:
/home/linuxbrew/.linuxbrew/include/b64/encode.h:25:31: error: ‘BUFFERSIZE’ was not declared in this scope
   encoder(int buffersize_in = BUFFERSIZE)
```

After looking at some diffs and commit history I found it was changed in this commit (sorry I can't link to a specific file like on github) https://sourceforge.net/p/libb64/git/ci/9a462ed91d9146b6aacf6a49f08c102b88fe75e7/

The buffer size of 4096 was changed to a definition BUFFERSIZE, so in order to get DCCL to compile I added this ```-DCMAKE_CXX_FLAGS=-DBUFFERSIZE=4096```.

It might need to be added to the project's cmake file so other people do not run into this issue. As an aside, is it possible to remove this dependency and replace it with something from a stdlib? The lib is quite old and not really maintained any more.

Cheers,

Russ

--- a/src/binary.h
+++ b/src/binary.h
@@ -31,6 +31,8 @@
 #define DCCL_HAS_B64 @DCCL_HAS_B64@
 
 #if DCCL_HAS_B64
+#include <cstdio>
+#define BUFFERSIZE BUFSIZ
 #include "b64/encode.h"
 #include "b64/decode.h"
 #endif

